lib/goth.ex

```diff
@@ -221,6 +224,40 @@ defmodule Goth do
     Goth.HTTPClient.init(config)
   end
 
+  defp put_default_source_credentials({:default, opts}) do
+    case Config.get(:token_source) do
+      {:ok, :oauth_jwt} ->
+        {:ok, private_key} = Config.get(:private_key)
+        {:ok, client_email} = Config.get(:actor_email)
+
+        credentials = %{
+          "private_key" => private_key,
+          "client_email" => client_email
+        }
+
+        {:service_account, credentials, opts}
+
+      {:ok, :oauth_refresh} ->
+        {:ok, refresh_token} = Config.get(:refresh_token)
+        {:ok, client_id} = Config.get(:project_id)
+        {:ok, client_secret} = Config.get(:client_secret)
+
+        credentials = %{
+          "refresh_token" => refresh_token,
+          "client_id" => client_id,
+          "client_secret" => client_secret
+        }
+
+        {:refresh_token, credentials, opts}
+
+      {:ok, :metadata} ->
+        {:metadata, opts}
+    end
+  end
+
+  defp put_default_source_credentials({type, credentials}), do: {type, credentials, []}
```

Comment:
Added this one when someone forgets to add the third element into the `source` tuple, but we can remove if doesn't make sense atm